src/xgb-classifier/xgboost-train.py

```diff
@@ -0,0 +1,137 @@
+from pathlib import Path
+from xgboost import XGBClassifier
+import logging
+
+# Setup logging
+logging.basicConfig(level=logging.INFO, format='%(asctime)s - %(levelname)s - %(message)s')
+logger = logging.getLogger(__name__)
+
+def train_model():
+    """Train XGBoost model for CVE severity classification"""
+    logger.info("Loading training dataset...")
+    
+    # Path to training dataset
+    data_path = 'sample_data/sample_data.csv'
+    
+    df = pd.read_csv(data_path)
+    logger.info(f"Loaded {len(df)} CVEs from {data_path}")
+    
+    # Check required columns
+    required_columns = ['cve_id', 'severity_label', 'severity_numeric']
+    missing_columns = [col for col in required_columns if col not in df.columns]
+    
+    if missing_columns:
+        logger.error(f"Missing required columns: {missing_columns}")
+        logger.info(f"Available columns: {list(df.columns)}")
+        return
+    
+    # Convert boolean features to integers (0/1) if needed
+    feature_columns = [col for col in df.columns if col not in ['cve_id', 'severity_label', 'severity_numeric']]
+    
+    logger.info(f"Found {len(feature_columns)} feature columns")
+    
+    # Handle any non-numeric features
+    for col in feature_columns:
+        if df[col].dtype == 'object':
+            logger.info(f"Converting column '{col}' from object to numeric")
+            df[col] = pd.to_numeric(df[col], errors='coerce').fillna(0)
+        df[col] = df[col].astype(int)
+    
+    # Remove any rows with missing target values
+    initial_count = len(df)
+    df = df.dropna(subset=['severity_numeric'])
+    if len(df) < initial_count:
+        logger.info(f"Removed {initial_count - len(df)} rows with missing severity_numeric values")
+    
+    # Prepare features and target
+    X = df[feature_columns]
+    y = df['severity_numeric'].astype(int)
+    
+    logger.info(f"Features: {len(feature_columns)} columns")
+    logger.info(f"Training samples: {len(X)}")
+    logger.info(f"Target distribution:\n{y.value_counts().sort_index()}")
+    
+    # Check if we have enough samples for each class
+    class_counts = y.value_counts()
+    if len(class_counts) < 2:
+        logger.warning("Only one class found in target variable!")
+    
+    # Train XGBoost model
+    logger.info("Training XGBoost model...")
+    
+    model = XGBClassifier(
+        n_estimators=200,
+        max_depth=6,
+        learning_rate=0.1,
+        random_state=42,
+        eval_metric='mlogloss',
+        verbosity=1,
+        class_weight='balanced'
+    )
+    
+    try:
+        model.fit(X, y)
+        logger.info("Training complete!")
+    except Exception as e:
+        logger.error(f"Training failed: {e}")
+        return
+    
+    # Create models directory
+    models_dir = Path('models')
+    models_dir.mkdir(exist_ok=True)
+    logger.info(f"Created models directory: {models_dir.absolute()}")
+    
+    # Save the model
+    model_path = models_dir / 'cve_severity_model.pkl'
```

Comment:
it would be good to push any saved models to a `src/aegis_ai_ml/etc/models` directory